lib/ngx/re.lua

```diff
@@ -84,6 +84,9 @@ end
 
 
 function _M.split(subj, regex, opts, ctx, max, res)
+    if subj == nil then
+        return nil, nil
```

Comment:
@gnought Well, I prefer make it undocumented. Anything documented has to be ensured backward compatibility forever. It would be the law.